File: Data Structures & Algorithms/sqrtx/submission-0.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left = 1
        right = x 

        while left + 1 < right:
            mid = (left + right) // 2

            if mid * mid == x:
                return mid 
            elif mid * mid < x:
                left = mid
            else:
                right = mid
        if left * left <= x:
            return left 
        if right * right <= x:
            return right

        return 

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == nullptr){
            return 0;
        }

        if (root->left == nullptr && root->right == nullptr){
            return 1;
        }

        int left_depth = 1  + maxDepth(root->left);
        int right_depth = 1 + maxDepth(root->right);

        return left_depth < right_depth ? right_depth: left_depth;

    }
};
